### ISSUE TITLE
Butchering Crash

### ISSUE THREAD

**Issue by Acalystus** [CleverRaven/Cataclysm-DDA#32695]
# Describe the bug

Every time I attempt to butcher  zombie corpses, the game crashes.

# Steps To Reproduce

Attempting to butcher even a single zombie corpse causes the game to crash. 

# Expected behavior

I expect it to butcher the zombies into the piles of fat and tainted meat.  It does this, but immediately crashes afterwards.

# Screenshots
![image](https://user-images.githubusercontent.com/53403255/62008975-a5c41a00-b116-11e9-8b63-34a5afc962da.png)

# Versions and configuration
- OS: Windows
    - OS Version: 10.0 1803
- Game Version: 0.D-5942-g666b2ec [64-bit]
- Graphics Version: Tiles
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Magiclysm [magiclysm],
    Mutant NPCs [mutant_npcs],
    Mythical Martial Arts [MMA],
    More Locations [more_locations],
    Fuji's More Buildings [FujiStruct],
    Urban Development [Urban_Development],
    More City Locations [cityside],
    Stats Through Kills [stats_through_kills]

# Additional context
[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3439298/crash.log)
[CDDA Save.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3439302/CDDA.Save.zip)

The saved game provided has several corpses near by that I've tried to butcher with no luck.

**Comment by anothersimulacrum** [comment]
Are you sure you uploaded the right save? This is what I get when I load it - no corpses in sight.

![catasave](https://user-images.githubusercontent.com/42699974/62009156-8e535400-b14b-11e9-95c2-4eeeab5bfb96.png)

**Comment by Acalystus** [comment]
![image](https://user-images.githubusercontent.com/53403255/62009399-f0946080-b11b-11e9-95c1-6de66358c647.png)

This is the game I just loaded.  There's several under the NPC and one in the corner to the south east.

**Comment by anothersimulacrum** [comment]
Huh - not getting an corpses. However, the save I load has the door the NPC is in closed, the npc is two tiles to the left, and the windows are open. Could you upload it again?

**Comment by Acalystus** [comment]
[The Living Lands.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3439325/The.Living.Lands.zip)

Sure.  This one might look a little different that the picture as I moved a bit after I loaded it to try something else.

**Comment by anothersimulacrum** [comment]
Still no corpses, but I did trigger a crash, thanks!
Here's a stack trace.
> The program has crashed.
See the log file for a stack trace.
CRASH LOG FILE: ./config/crash.log
VERSION: 0.D-5942-g666b2ec9ec-dirty
TYPE: Signal
MESSAGE: SIGSEGV: Segmentation fault
STACK TRACE:

	./cataclysm-tiles(_Z21debug_write_backtraceRSo+0x42) [0x55ae90aca716]
	./cataclysm-tiles(+0x4acc0d) [0x55ae90abbc0d]
	./cataclysm-tiles(+0x4ace23) [0x55ae90abbe23]
	/usr/lib/libc.so.6(+0x3a7e0) [0x7f3c775f67e0]
	./cataclysm-tiles(_ZN4item18get_safe_referenceEv+0x9) [0x55ae90c0068d]
	./cataclysm-tiles(_ZN13item_locationC1ERK10map_cursorP4item+0x47) [0x55ae90c5e351]
	./cataclysm-tiles(_ZNK3map26get_active_items_in_radiusERK8tripointiNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x252) [0x55ae90d0e8c4]
	./cataclysm-tiles(_ZN3npc19find_corpse_to_pulpEv+0x14f) [0x55ae90eaa781]
	./cataclysm-tiles(_ZN3npc13address_needsEf+0x375) [0x55ae90eaac6b]
	./cataclysm-tiles(_ZN3npc4moveEv+0x636) [0x55ae90eb2090]
	./cataclysm-tiles(_ZN4game7monmoveEv+0xb4f) [0x55ae90b6a0ab]
	./cataclysm-tiles(_ZN4game7do_turnEv+0x850) [0x55ae90b71542]
	./cataclysm-tiles(main+0x1241) [0x55ae90922957]
	/usr/lib/libc.so.6(__libc_start_main+0xf3) [0x7f3c775e2ee3]
	./cataclysm-tiles(_start+0x2e) [0x55ae9097fcce]

	Attempting to repeat stack trace using debug symbols...
	debug_write_backtrace(std::ostream&)
	.../src/debug.cpp:621
	std::__cxx11::basic_ostringstream<char, std::char_traits<char>, std::allocator<char> >::str() const
	/usr/include/c++/9.1.0/sstream:678
	log_crash
	.../src/crash.cpp:263
	signal_handler
	.../src/crash.cpp:292
	??
	??:0
	std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count(std::__shared_count<(__gnu_cxx::_Lock_policy)2> const&)
	/usr/include/c++/9.1.0/bits/shared_ptr_base.h:734
	std::__shared_ptr<item, (__gnu_cxx::_Lock_policy)2>::__shared_ptr<safe_reference_anchor::empty>(std::__shared_ptr<safe_reference_anchor::empty, (__gnu_cxx::_Lock_policy)2> const&, item*)
	/usr/include/c++/9.1.0/bits/shared_ptr_base.h:1164
	std::shared_ptr<item>::shared_ptr<safe_reference_anchor::empty>(std::shared_ptr<safe_reference_anchor::empty> const&, item*)
	/usr/include/c++/9.1.0/bits/shared_ptr.h:235
	safe_reference<item> safe_reference_anchor::reference_to<item>(item*)
	.../src/safe_reference.h:66
	item::get_safe_reference()
	.../src/item.cpp:271
	item_location::impl::item_on_map::item_on_map(map_cursor const&, item*)
	.../src/item_location.cpp:160
	item_location::item_location(map_cursor const&, item*)
	.../src/item_location.cpp:445
	void std::__cxx11::list<item_location, std::allocator<item_location> >::_M_insert<map_cursor, item*>(std::_List_iterator<item_location>, map_cursor&&, item*&&)
	/usr/include/c++/9.1.0/bits/stl_list.h:1908
	void std::__cxx11::list<item_location, std::allocator<item_location> >::emplace_back<map_cursor, item*>(map_cursor&&, item*&&)
	/usr/include/c++/9.1.0/bits/stl_list.h:1223
	map::get_active_items_in_radius(tripoint const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) const
	.../src/map.cpp:8392
	std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()
	/usr/include/c++/9.1.0/bits/basic_string.h:658 (discriminator 4)
	npc::find_corpse_to_pulp()
	.../src/npcmove.cpp:2973 (discriminator 4)
	npc::address_needs(float)
	.../src/npcmove.cpp:1759
	npc::move()
	.../src/npcmove.cpp:765
	game::monmove()
	.../src/game.cpp:4269
	game::do_turn()
	.../src/game.cpp:1522
	main
	.../src/main.cpp:688
	__libc_start_main
	??:?
	_start
	??:?

**Comment by l29ah** [comment]
Got the same crash it seems:
```
	cataclysm(_Z21debug_write_backtraceRSo+0x3a) [0x5580183afe4a]
	cataclysm(+0x6880f2) [0x5580183990f2]
	cataclysm(+0x688558) [0x558018399558]
	/lib64/libc.so.6(+0x396b0) [0x7fb720f386b0]
	cataclysm(_ZN4item18get_safe_referenceEv+0xb) [0x5580185dc0ab]
	cataclysm(_ZN13item_locationC1ERK10map_cursorP4item+0x46) [0x558018686d26]
	cataclysm(_ZNK3map26get_active_items_in_radiusERK8tripointiNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE+0x28a) [0x5580187b7b6a]
	cataclysm(_ZN3npc19find_corpse_to_pulpEv+0xff) [0x558018a9316f]
	cataclysm(_ZN3npc13address_needsEf+0x2dd) [0x558018a9c5cd]
	cataclysm(_ZN3npc4moveEv+0xb55) [0x558018aa57f5]
	cataclysm(_ZN4game7monmoveEv+0xa52) [0x5580184efb22]
	cataclysm(_ZN4game7do_turnEv+0x998) [0x5580184f0f98]
	cataclysm(main+0xcf9) [0x558018120c99]
	/lib64/libc.so.6(__libc_start_main+0xeb) [0x7fb720f22ecb]
	cataclysm(_start+0x2a) [0x558018174a0a]
```

**Comment by hansonry** [comment]
I am having a similar issue. I just pulled down today. I have a save where I attempt to butcher a beaver in the river "Full Butchery". I built with g++-9 on Linux Mint. 
[Boyne Falls.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/3603092/Boyne.Falls.zip)

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3603088/crash.log)

**Comment by hansonry** [comment]
I got it fixed on my local machine. I created a pull request here:
https://github.com/CleverRaven/Cataclysm-DDA/pull/33976

**Comment by ZhilkinSerg** [comment]
Fixed in #33976.
